web/onos-gui/src/app/app.component.ts

```diff
@@ -42,20 +56,68 @@ export class AppComponent {
   }
 
   listChanges() {
-    this.diags.requestChanges();
+    this.changes.length = 0;
+    this.diags.requestChanges((change: Change) => {
+      this.changes.push(change);
+    });
   }
 
   listDeviceSummary() {
-    this.admin.requestSummary();
+    this.admin.requestSummary((e: grpcWeb.Error, r: DeviceSummaryResponse) => {
+        if (e != null) {
+          console.warn(e.code, e.message);
+          return;
+        }
+        this.deviceCount = r.getCount()
+    });
   }
 
-  listGnmiCapabilities() {
+  listGnmiCapabilities(): void {
     this.gnmi.requestCapabilities((e: grpcWeb.Error, r: CapabilityResponse) => {
       if (e != null) {
-        console.log('Error getting capabilities from onos-config');
-        console.log(e.code, e.message);
+        console.warn('Error getting capabilities from onos-config');
+        console.warn(e.code, e.message);
       }
       this.capabilites = r;
     });
   }
+
+  getConfigByDevice(deviceId: string): void {
+    this.gnmi.requestGetAllByDevice(deviceId, (e, r) => this.handleGetResponse(e, r));
+  }
+
+  /**
+   * Callback for handling the GetResponse
+   * @param e error
+   * @param r GetResponse
+   */
+  handleGetResponse(e: grpcWeb.Error, r: GetResponse): void {
+    console.log('GetResponse received');
+    if (e != null) {
+      console.warn('Error getting capabilities from onos-config');
+      console.warn(e.code, e.message);
+      return;
+    }
+    const notifications: Array<Notification> = r.getNotificationList();
+    this.updates.length = 0;
+    for (const n of notifications) {
+      for (const u of n.getUpdateList()) {
+        console.log('Update', u.getVal().getValueCase());
+        if (u.getVal().getValueCase() == TypedValue.ValueCase.JSON_VAL) {
+          const jsonValue = new TextDecoder("utf-8").decode(u.getVal().getJsonVal_asU8());
+          const jsonUpdate = <UpdateAsString>{
+            path: "/",
+            value: jsonValue,
+          };
+          this.updates.push(jsonUpdate);
+          console.log('Update', jsonUpdate);
+        }
+      }
+    }
+    console.log('GetResponse finished');
+  }
+
+  pathToString(p: Path): string {
```

Comment:
is this method going to end up as complicated as the ones we have in onos-config are are we going to skip dealing with namespaces and keys here ?

Reply:
good question - i'm not sure if I'm going to turn them in to paths like we have internally

Reply:
if there is some son library we could ask for the elements as json values and reuse that. I'd avoid complicating this very much.

Reply:
it wasn't being used. will have to add something at some stage in the future